# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: security/sandbox/win/src/remotesandboxbroker/RemoteSandboxBrokerParent.h
# Commit: 87a66c9bce6e
# Full Hash: 87a66c9bce6e60c232336ac5b64238befbead24f
# Author: Jean-Yves Avenard <jyavenard@mozilla.com>
# Date: 2020-07-01 21:32:12
# Description:
#   Bug 1648326 - Don't dispatch runnable on the running taskqueue. r=bobowen,jld
#   
#   The current taskqueue is blocked until the current function has finished; Running the event loop would only process events on the running thread.
#   
#   Additionally, we make mIPCLaunchThread an nsISerialEventTarget to guarantee that at shutdown the tasks are run in order regardless of the IPC Launch Thread type.
# ==============================================================================

diff -r 1c8faab05606 -r 87a66c9bce6e security/sandbox/win/src/remotesandboxbroker/RemoteSandboxBrokerParent.h
--- a/security/sandbox/win/src/remotesandboxbroker/RemoteSandboxBrokerParent.h	Tue Jun 30 16:03:02 2020 +0000
+++ b/security/sandbox/win/src/remotesandboxbroker/RemoteSandboxBrokerParent.h	Wed Jul 01 06:46:59 2020 +0000
@@ -26,7 +26,9 @@
   // Asynchronously launches the launcher process.
   // Note: we rely on the caller to keep this instance alive
   // until this promise resolves.
-  RefPtr<GenericPromise> Launch(const nsTArray<uint64_t>& aHandlesToShare);
+  // aThread is the thread to use to resolve the promise on if needed.
+  RefPtr<GenericPromise> Launch(const nsTArray<uint64_t>& aHandlesToShare,
+                                nsISerialEventTarget* aThread);
 
  private:
   void ActorDestroy(ActorDestroyReason aWhy) override;